video/out/hwdec/hwdec_drmprime.c

```diff
@@ -167,7 +167,10 @@ static int mapper_init(struct ra_hwdec_mapper *mapper)
     struct dmabuf_interop_priv *p = mapper->priv;
 
     mapper->dst_params = mapper->src_params;
-    mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
+    if (mapper->src_params.hw_subfmt == IMGFMT_420P)
+    	mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
+    else
+    	mapper->dst_params.imgfmt = IMGFMT_NV12;
```

Comment:
Is this a way to get SAND128 to be treated as NV12 without needing to have an IMGFMT for it?

My main concern is that in general, there can be many more than two possible formats handled here. If you can do additional checks against the format descriptor (number of bits, planes, chroma subsampling, etc) to narrow down the case, it'll be less concerning.

Reply:
Okay the problem with the descriptors you gave as an example, is that those are generated by ra_get_imgfmt_desc which fails for RPI8 and 10 (this problem is pretty much how this all started actually).

That said, here are my ideas to check it without just dumbly grouping else as NV12, let me know which sounds the best.

Check imgfmt is equal to drm_prime,
```
    if (mapper->src_params.hw_subfmt != IMGFMT_420P && mapper->src_params.imgfmt == IMGFMT_DRMPRIME)
    	mapper->dst_params.imgfmt = IMGFMT_NV12;
    else
    	mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
```
check hw_subfmt integer is in the ffmpeg range,
```
    if (mapper->src_params.hw_subfmt > IMGFMT_AVPIXFMT_START && mapper->src_params.hw_subfmt < IMGFMT_AVPIXFMT_END)
    	mapper->dst_params.imgfmt = IMGFMT_NV12;
    else
    	mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
```
check hw_subfmt string is rpi4_8 or rpi4_10,
```
    if (strcmp(mp_imgfmt_to_name(mapper->src_params.hw_subfmt), "rpi4_8") == 0 || strcmp(mp_imgfmt_to_name(mapper->src_params.hw_subfmt), "rpi4_10") == 0)
    	mapper->dst_params.imgfmt = IMGFMT_NV12;
    else
    	mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
```
analyze other mp_image_params,
```
    if (mapper->src_params.space == 2 && so on...)
    	mapper->dst_params.imgfmt = IMGFMT_NV12;
    else
    	mapper->dst_params.imgfmt = mapper->src_params.hw_subfmt;
```
or some combination of these.

Here is some of the data I collected about the two formats, one from a vaapi system and one from the pi, both using the same HEVC 10-bit file.

NV12 output:
```
imgfmt: vaapi
hw_subfmt: nv12
w: 128
h: 128
p_w: 0
p_h: 1
space: 0
levels: 0
primaries: 0
gamma: 0
light: 0
sig_peak: 0.000000
chroma_location: 0
rotate: 0
stereo3d: 0
alpha: 0
```
RPI output:
```
imgfmt: drm_prime
hw_subfmt: rpi4_10
w: 1920
h: 1080
p_w: 1
p_h: 1
space: 2
levels: 1
primaries: 3
gamma: 1
light: 1
sig_peak: 1.000000
chroma_location: 2
rotate: 0
stereo3d: 0
alpha: 0
```

Reply:
Ah. Then definitely do the string comparison. Just add a comment there explaining it's a string comparison because the pix fmt isn't in upstream ffmpeg and therefore not stable.